seed/building_sync/mappings.py

```diff
@@ -0,0 +1,1005 @@
+    """
+    # get the element from the schema
+    element_path = tree.getpath(element)
+    schema_element = schema.find(element_path)
+    if schema_element is None:
+        raise Exception(f'Unable to find path in schema: "{element_path}"')
+
+    ordered_children = [child.name for child in schema_element.iterchildren()]
+
+    # construct a function for sorting an element's children by returning the index of the
+    # child according to the ordering of the schema
+    def _getkey(element):
+        if isinstance(element, etree._Comment):
+            # put comments at the end
+            return 100
+        elif not isinstance(element, etree._Element):
+            raise Exception(f'Unknown type while sorting: "{type(element)}"')
+
+        if element.tag not in ordered_children:
+            # a more helpful exception that the one raised by .index()
+            raise Exception(f'Failed to find "{element.tag}" in {ordered_children}')
+
+        return ordered_children.index(element.tag)
+
+    return _getkey
+
+
+def update_element(element, target, value):
+    """updates an element's text or attribute with a value
+
+    :param element: lxml.Element, element to update
+    :param target: string, an attribute as "@<attribute>" or "text"
+    :param value: string, value to set for target
+    """
+    if target.startswith('@'):
+        element.set(target[1:], value)
+    elif target == 'text':
+        element.text = value
+    else:
+        raise Exception(f'Unrecognized target "{target}"')
+
+
+def update_tree(schema, tree, xpath, target, value, namespaces):
+    """updates the tree at the xpath with the given value
+
+    :param schema: xmlschema.XmlSchema, schema for determining sequence ordering
+    :param tree: lxml.ElementTree, tree to update
+    :param xpath: string, absolute xpath to the element to update (i.e. starts with /auc:BuildingSync/...)
+    :param target: string, an attribute, "@<attribute>", or "text" to indicate what to update on the element
+    :param value: string, the value to set for that node
+    """
+    last_element, xpath_remainder = find_last_in_xpath(tree, xpath, namespaces)
+
+    if xpath_remainder:
+        # build out the remaining path and then update the value
+        new_element = build_path(last_element, xpath_remainder)
+        update_element(new_element, target, value)
+
+        # since we added new elements, we need to reorder at the base of our addition
+        getkey = children_sorter_factory(schema, tree, last_element)
+        last_element[:] = sorted(last_element, key=getkey)
+    else:
+        # element already exists, update its value
+        update_element(last_element, target, value)
+
+
+# Base mapping for BuildingSync schema version 2.0-pr1
+# NOTE: this is actually a mix of 2.0-pr1 and 1.0 as it uses 'TimeStamp' rather than 'Timestamp'
+# and it allows the 'Reports' element to exist or not exist.
+BASE_MAPPING_V2_PR1 = {
```

Comment:
What do you think about splitting these mappings out into their own files for organization purposes?

Reply:
Agreed, though holding off for now